src/nvim/os/stdpaths.c

```diff
@@ -93,6 +96,46 @@ bool appname_is_valid(void)
   return true;
 }
 
+/// Remove duplicate directories in the given XDG directory.
+/// @param[in]  List of directories possibly with duplicates
+/// @param[out]  List of directories without duplicates
+static char *xdg_remove_duplicate(char *ret, const char *sep)
+{
+  kvec_t(char *) data = KV_INITIAL_VALUE;
+  char *saveptr;
+
+  char *token = os_strtok(ret, sep, &saveptr);
+  while (token != NULL) {
+    // Check if the directory is not already in the list
+    bool is_duplicate = false;
+    for (size_t i = 0; i < data.size; i++) {
+      if (striequal(kv_A(data, i), token)) {
```

Comment:
I don't think this is correct. It should use `path_fnamecmp()`.

Reply:
got it